packages/react-data-grid/src/RowsContainer.js

```diff
@@ -1,35 +1,33 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const SimpleRowsContainer = (props) => {
-  return (
-    <div key="rows-container">
-      {props.rows}
-    </div>
-  );
-};
+export const DEFAULT_CONTEXT_MENU_ID = 'rgdContextMenu';
+
+const SimpleRowsContainer = (props) => <div key="rows-container">{props.rows}</div>;
 
 SimpleRowsContainer.propTypes = {
   width: PropTypes.number,
   rows: PropTypes.array
 };
 
+export const getNewContextMenuProps = ({ contextMenuId, rowIdx, idx }) => ({
+  rowIdx, idx, id: contextMenuId || DEFAULT_CONTEXT_MENU_ID
+});
+
 class RowsContainer extends React.Component {
   constructor(props) {
     super(props);
     this.plugins = props.window ? props.window.ReactDataGridPlugins : window.ReactDataGridPlugins;
     this.hasContextMenu = this.hasContextMenu.bind(this);
     this.renderRowsWithContextMenu = this.renderRowsWithContextMenu.bind(this);
-    this.getContextMenuContainer = this.getContextMenuContainer.bind(this);
-    this.state = {ContextMenuContainer: this.getContextMenuContainer(props)};
+    this.validateContextMenu = this.validateContextMenu.bind(this);
+
+    this.validateContextMenu();
```

Comment:
I wonder if `constructor` is the right place for validation. Did not find any recommendation on where to throw the errors but I think `render` method would probably be a better place. It is recommended to avoid any side effects in constructor 
https://reactjs.org/docs/react-component.html#constructor

```
render() {
    if (this.hasContextMenu())  {
        if (!this.plugins) { throw new Error('...') }
        return this.renderRowsWithContextMenu();
    }
    return <SimpleRowsContainer {...this.props} />;
}
```

@malonecj @diogofcunha thoughts?

Reply:
@amanmahajan7 we shouldn't add any side effects in the constructor even when not using react, creating a new object should never affect other parts of an application, you are correct, in react the constructor is used only to set the initial state